### PR TITLE
MWPW-131134 - Marquee: media-credit container text disappears if italicized

### PR DESCRIPTION
**Pull request by williambsm**
* Allow text formatting of the media credit section

Resolves: [MWPW-113134](https://jira.corp.adobe.com/browse/MWPW-131134)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.page/drafts/00lightning/bugs/substance-magazine-article-marquee-image-credit-disappears-italics?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/00lightning/bugs/substance-magazine-article-marquee-image-credit-disappears-italics?milolibs=mwpw-113134--milo--wbstry&martech=off

I'm not entirely sure that this the optimal solution, advise is much appreciated, this work around prevents the primary image from being stolen into the media credit container.

From my understanding, the test file doesn't need to be updated, as it still contains the same element no matter the outcome. Please advise me if otherwise!

### CODE REVIEW

**Review by auniverseaway — CHANGES_REQUESTED** [review]
Please update the tests.

**Comment by honstar** [comment]
@williambsm : what's the current status of this PR? Chris requested changes to tests that are currently failing. Also, there are merge conflicts that need to be addressed. Thx

**Review by auniverseaway — APPROVED** [review]

**Comment by williambsm** [comment]
> @williambsm : what's the current status of this PR? Chris requested changes to tests that are currently failing. Also, there are merge conflicts that need to be addressed. Thx

I have just solved the merge conflict, and the tests were updated appropriately. It is ready for review again although already approved by Chris, we should be good to merge!

**Comment by williambsm** [comment]
cc @honstar

File: libs/blocks/marquee/marquee.js [hunk 1]
```diff
@@ -161,8 +161,16 @@ export default function init(el) {
...
     }
+
+    let mediaCreditInner;
+
     if (media?.lastChild.textContent.trim()) {
-      const mediaCreditInner = createTag('p', { class: 'body-s' }, media.lastChild.textContent);
+      mediaCreditInner = createTag('p', { class: 'body-s' }, media?.lastChild.textContent.trim());
+    } else if (media.lastElementChild && media.lastElementChild.tagName !== 'PICTURE') {
+      mediaCreditInner = media.lastElementChild;
+    }
```
[the remark this answers is not in the document]
Comment (williambsm) [reply libs/blocks/marquee/marquee.js]: Ha, I also preferred this version but I am taking a very "don't touch what I don't need to" approach. 

It's fixed now.

File: libs/blocks/marquee/marquee.js [hunk 2]
```diff
@@ -161,8 +161,16 @@ export default function init(el) {
...
       foreground.insertAdjacentElement('beforebegin', media);
     }
-    if (media?.lastChild.textContent.trim()) {
-      const mediaCreditInner = createTag('p', { class: 'body-s' }, media.lastChild.textContent);
+
+    let mediaCreditInner;
+    const txtContent = media?.lastChild.textContent.trim();
+    if (txtContent) {
+      mediaCreditInner = createTag('p', { class: 'body-s' }, txtContent);
+    } else if (media.lastElementChild && media.lastElementChild.tagName !== 'PICTURE') {
```
Comment (overmyheadandbody) [inline libs/blocks/marquee/marquee.js:169]: Maybe shorten this to just `media.lastElementChild?.tagName !== 'PICTURE'`?

**Review by overmyheadandbody — APPROVED** [review]

**Review by honstar — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Comment by rgclayton** [comment]
@williambsm How are we looking here? Is this ready to be merged?

File: libs/blocks/marquee/marquee.js (hunk 2 above)
Comment (williambsm) [reply libs/blocks/marquee/marquee.js]: Adressed, thank you.

**Comment by williambsm** [comment]
> @williambsm How are we looking here? Is this ready to be merged?

Yup! All is ready.

**Comment by hparra** [comment]
This was not ready for merge. Failed check failures, including consistent UT failure: https://github.com/adobecom/milo/pull/843/checks

@williambsm Can we revert this if can't fix within hour?

**Comment by hparra** [comment]
Mitigated with https://github.com/adobecom/milo/pull/1006
